app/streams/publishing_api/message_handler.rb

```diff
@@ -32,8 +32,17 @@ def is_links_update?
   end
 
   def links_have_changed?
-    current_links = old_item.raw_json['expanded_links'].deep_sort
-    new_links = new_item.raw_json['expanded_links'].deep_sort
+    current_links = if old_item && old_item.raw_json['expanded_links']
```

Comment:
Could you please add a method in the Item to return the expanded links. You could unit test it and this method can be simplified to:

```ruby
HashDiff::Comparison.new(
  current_links.expanded_links.deep_sort,  
  new_links.expanded_links.deep_sort
).diff.present?
```